src/cpp/rtps/builtin/data/ParticipantProxyData.cpp

```diff
@@ -452,6 +452,68 @@ bool ParticipantProxyData::updateData(ParticipantProxyData& pdata)
     return true;
 }
 
+void ParticipantProxyData::set_persistence_guid(const GUID_t & guid)
+{
+    // only valid values
+    if(guid == c_Guid_Unknown)
+    {
+        return;
+    }
+
+    // generate pair
+    std::pair<std::string, std::string> persistent_guid;
+    persistent_guid.first = "PID_PERSISTENCE_GUID";
+
+    std::ostringstream data;
+    data << guid;
+    persistent_guid.second = data.str();
+
+    // if exists replace
+    std::vector<std::pair<std::string, std::string>> & props = m_properties.properties;
+
+    std::vector<std::pair<std::string, std::string>>::iterator it =
+        std::find_if(
+            props.begin(),
+            props.end(),
+            [&persistent_guid](const std::pair<std::string, std::string> & p) {
```

Comment:
We should treat lambdas as if declaring a method at current indentation (i.e break line after parameters, and then indent the content accordingly.

Reply:
ok